nano/node/noderpcconfig.cpp

```diff
@@ -0,0 +1,77 @@
+#include <nano/node/noderpcconfig.hpp>
+
+#include <nano/lib/blocks.hpp>
+#include <nano/lib/config.hpp>
+#include <nano/lib/jsonconfig.hpp>
+#include <nano/lib/rpcconfig.hpp>
+
+nano::error nano::node_rpc_config::serialize_json (nano::jsonconfig & json) const
+{
+	json.put ("version", json_version ());
+	json.put ("enable_sign_hash", enable_sign_hash);
+	json.put ("max_work_generate_difficulty", nano::to_string_hex (max_work_generate_difficulty));
+	json.put ("rpc_path", rpc_path);
+	json.put ("rpc_in_process", rpc_in_process);
+	return json.get_error ();
+}
+
+nano::error nano::node_rpc_config::deserialize_json (bool & upgraded_a, nano::jsonconfig & json, boost::filesystem::path const & data_path)
+{
+	auto version_l (json.get_optional<unsigned> ("version"));
+	if (!version_l)
+	{
+		version_l = 1;
+		json.put ("version", *version_l);
+		json.erase ("frontier_request_limit");
+		json.erase ("chain_request_limit");
+
+		// Don't migrate enable_sign_hash as this is not needed by the external rpc process, but first save it.
+		json.get_optional ("enable_sign_hash", enable_sign_hash, false);
+
+		json.erase ("enable_sign_hash");
+		json.erase ("max_work_generate_difficulty");
+
+		migrate (json, data_path);
+
+		json.erase ("io_threads");
+
+		json.put ("enable_sign_hash", enable_sign_hash);
+		json.put ("max_work_generate_difficulty", nano::to_string_hex (max_work_generate_difficulty));
+		json.put ("rpc_path", get_default_rpc_filepath ());
+		bool rpc_in_process_l;
+		json.get_optional<bool> ("rpc_in_process", rpc_in_process_l);
+		if (!rpc_in_process)
```

Comment:
I noticed `rpc_in_process` doesn't end up in the upgraded json file. If that was the intention here, maybe use the single-parameter `get_optional` overload which returns an boost::optional. If a value is not set, then write it out.

Reply:
Yeh was using the wrong constructs, thanks.